torch_em/loss/wrapper.py

```diff
@@ -43,30 +44,65 @@ def forward(self, prediction, target, **kwargs):
 
 
 class ApplyMask:
-    def __call__(self, prediction, target, mask):
-        mask.requires_grad = False
+    def _crop(prediction, target, mask, channel_dim):
+        mask = mask.type(torch.bool)
+        # TODO: make this work for different
+        # masks per channel
+        assert mask.shape[channel_dim] == 1
```

Comment:
The case where we have a full mask is complicated and I can't see an option right now to make it work where we also preserve the N x C structure. I think we can go ahead with the current solution, but should throw a more meaningful error message here. I suggest to throw a `ValueError`, which should also indicate that using `masking_method=multiply` will fix the issue.

Reply:
done